code/modules/mining/mine_turfs.dm

```diff
@@ -381,6 +383,16 @@
 		var/randumb = pickweight(mob_spawn_list)
 		new randumb(T)
 	return
+/turf/open/floor/plating/asteroid/airless/cave/proc/SpawnFlora(turf/T)
+	if(prob(12))
+		if(istype(loc, /area/mine/explored) || istype(loc, /area/lavaland/surface/outdoors/explored))
+			return
+		for(var/obj/structure/flora/ash/F in urange(3,T)) //Allows for growing patches, but not ridiculous stacks of flora
+			if(istype(F, /obj/structure/flora/ash))
+				return
+		var/randumb = pickweight(flora_spawn_list)
```

Comment:
I would advise making this code the following, so that patches of a type of flora can show up.

    var/randumb = pickweight(flora_spawn_list)
    for(var/obj/structure/flora/ash/F in urange(3,T))
        if(!istype(F, randumb))
            return
    new randumb(T)